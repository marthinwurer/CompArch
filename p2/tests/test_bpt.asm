;
; tests BPT
;
; uses instructions MOV, HALT
;
; uses addressing modes 0, 6
;
	.globl	start
start:	bpt
	mov	x,r5
	bpt
	halt
;
x:	.word	42
	.end	start
