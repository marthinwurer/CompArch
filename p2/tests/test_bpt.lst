      1                                 ;
      2                                 ; tests BPT
      3                                 ;
      4                                 ; uses instructions MOV, HALT
      5                                 ;
      6                                 ; uses addressing modes 0, 6
      7                                 ;
      8 000000                                   .globl  start
      9 000000  000003                  start:   bpt
     10 000002  016705  000004                   mov     x,r5
     11 000006  000003                           bpt
     12 000010  000000                           halt
     13                                 ;
     14 000012  000042                  x:       .word   42
     15 000014  000000'                          .end    start


pc     = 000007 %  
r0     = 000000 %  
r1     = 000001 %  
r2     = 000002 %  
r3     = 000003 %  
r4     = 000004 %  
r5     = 000005 %  
r6     = 000006 %  
r7     = 000007 %  
sp     = 000006 %  
start    000000 RG 
x        000012 R  
