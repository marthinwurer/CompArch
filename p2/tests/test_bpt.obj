000 2 03 00
002 4 c5 1d 04 00
006 2 03 00
008 2 00 00
00a 2 22 00
000
