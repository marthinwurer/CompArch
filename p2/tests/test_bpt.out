CPU "ARCH" Simulator, 2.5a(Jan 22 2015)
-----------------------------------------

Memory sets starting address to 0
000000:  PS[00]  000003  BPT
  R0[000000]  R1[000000]  R2[000000]  R3[000000]
  R4[000000]  R5[000000]  R6[000000]  R7[000002]
000002:  PS[00]  016705 000004  MOV 000004(R7),R5
000006:  PS[00]  000003  BPT
  R0[000000]  R1[000000]  R2[000000]  R3[000000]
  R4[000000]  R5[000042]  R6[000000]  R7[000010]
000010:  PS[00]  000000  HALT

Machine Halted - HALT instruction

Simulated time 23 cycles

LAST CPUObject DESTROYED; END OF SIMULATION
